packages/descendants/README.md

```diff
@@ -0,0 +1,200 @@
+---
+last_updated: 2024-02-01
+---
+
+# Descendants
+
+![npm (scoped)](https://img.shields.io/npm/v/@leafygreen-ui/descendants.svg)
+
+#### [View on MongoDB.design](https://www.mongodb.design/component/descendants/example/)
+
+## Installation
+
+### Yarn
+
+```shell
+yarn add @leafygreen-ui/descendants
+```
+
+### NPM
+
+```shell
+npm install @leafygreen-ui/descendants
+```
+
+## Overview
+
+`Descendants` is an internal utility that allows components to track all rendered descendants.
+
+This is useful when developing menus in order to track when items are rendered/un-rendered.
+
+## Usage
+
+There are 4 steps required to set up a pair of components as Parent/Descendent.
+
+```tsx
+/**
+ * 1. Create a new Context
+ *
+ * We need to create a new Context object
+ * in order for the parent & child to know their relationships.
+ *
+ * Without this, a descendant won't know what parent context it belongs to.
+ * This also enables us to nest different descendant contexts
+ *
+ * e.g.
+ * A Menu that a second fly-out menu
+ * shouldn't track fly-out menu items as its own descendants)
+ *
+ */
+const MyDescendantsContext = createDescendantsContext('MyDescendantsContext');
+
+export const MyParent = ({ children, ...rest }: ComponentProps<'div'>) => {
+  /**
+   * 2. Initialize an empty descendants list and setter
+   *
+   * We call this _outside_ the Provider
+   * so we can access the `descendants` object
+   * from the Parent level.
+   */
+  const { descendants, dispatch } = useInitDescendants<HTMLDivElement>();
+
+  /**
+   * 3. Pass the context, descendants list and setter into the provider
+   *
+   * We need to pass the context value into the Provider
+   * in order for the Parent to have knowledge
+   * of its specific context
+   * (see fly-out menu example in step 1.)
+   */
+  return (
+    <DescendantsProvider
+      context={MyDescendantsContext}
+      descendants={descendants}
+      dispatch={dispatch}
+    >
+      <div {...rest}>{children}</div>
+    </DescendantsProvider>
+  );
+};
+
+export const TestDescendant = ({
+  children,
+  ...rest
+}: ComponentProps<'div'>) => {
+  /**
+   * 4. Establish a child component as a descendant
+   *
+   * Pass the context value into the hook
+   * in order to establish this element
+   */
+  const { index, ref } = useDescendant(TestDescendantContext);
```

Comment:
should this be the same context as used in the Parent? (I'd assume so) If so, let's use the same name

Reply:
Good catch